### PR TITLE
Make migrator usable with postgres

### PR DESCRIPTION
**Pull request by michal12334**

### CODE REVIEW

**Review by Dragemil — COMMENTED** [review]
Wouldn't here be value in providing a more specific base classes as well (inheriting from this base class) for our common cases - a SqlServer one, and a Postgres one?

File: src/Infrastructure/LeanCode.EFMigrator/BaseFactory.cs [hunk 1]
```diff
@@ -15,31 +15,24 @@ public abstract class BaseFactory<TContext, TFactory> : IDesignTimeDbContextFact
         typeof(TFactory).Assembly.GetName().Name
         ?? throw new InvalidOperationException("This type is not supported on Assembly-less runtimes.");
 
-    protected virtual void UseAdditionalSqlServerDbContextOptions(SqlServerDbContextOptionsBuilder builder) { }
-
     protected virtual void UseAdditionalDbContextOptions(DbContextOptionsBuilder<TContext> builder) { }
```
Comment (marcin-dardzinski) [inline src/Infrastructure/LeanCode.EFMigrator/BaseFactory.cs]: Do we need this, can't you do the same in `UseDbProvider`?

**Review by marcin-dardzinski — APPROVED** [review]

**Comment by jakubfijalkowski** [comment]
@Dragemil we would need separate assemblies for that, or reference both SqlServer and Postgres-specific packages in the migrator, no matter what db we use.

**Comment by Saancreed** [comment]
A bit off-topic for this but have we considered using [migration bundles](https://learn.microsoft.com/en-us/ef/core/managing-schemas/migrations/applying?tabs=dotnet-core-cli#bundles)? Maybe it would allow us to reduce our `Main` to just applying the seed if needed, or even get rid of it altogether if we don't.

**Review by Saancreed — APPROVED** [review]

**Comment by jakubfijalkowski** [comment]
I think we rejected it some time ago but I don't remember why we did that. I agree that we should look at it once again.

**Pull request merged**